x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java

```diff
@@ -176,6 +171,7 @@ public final class SamlRealm extends Realm implements Releasable {
     private final Supplier<EntityDescriptor> idpDescriptor;
 
     private final SpConfiguration serviceProvider;
+    private final SamlRealmSettings.UserAttributeNameConfiguration userAttributeNameConfiguration;
```

Comment:
Because the settings objects are now created every time they're used (due to the function type approach) I've created this to hold the attribute names used when creating a user.